Library_Engine/Query/Library.cs

```diff
@@ -272,6 +280,23 @@ private static void AddToPathDictionary(string path, string dictionaryPath)
                 AddToPathDictionary(basePath, dictionaryPath);
         }
 
+        /***************************************************/
+
+        private static void AddOldPaths(string path, string dictionaryPath)
+        {
+            Dictionary<string, List<string>> tolOldPaths = Versioning.Query.DatasetToOldPaths();
```

Comment:
This will be called on each recursion, but I don't see that the data would change? Could this be a variable input into the method instead that's passed along?